### ISSUE TITLE
Issues with dlsym

### ISSUE THREAD

**Issue by LinuxRocks2000** [bazelRio/bazelRio#110]
Our .bazelrc file is identical to the examples, and our WORKSPACE and BUILD files are only slightly modified for a custom library, but we keep getting a linker error relating to dlsym: 
```
/home/awesome/.cache/bazel/_bazel_awesome/3310ac2ce17be999eb013d710f5b6147/external/__bazelrio_roborio_toolchain_linux/frc2022/roborio/bin/../arm-frc2022-linux-gnueabi/usr/lib/gcc/arm-frc2022-linux-gnueabi/7.3.0/../../../../../../arm-frc2022-linux-gnueabi/bin/ld: external/__bazelrio_com_revrobotics_frc_revlib-driver_linuxathenastatic/linux/athena/static/libREVLibDriver.a(CANSparkMaxHeartbeat.o): undefined reference to symbol 'dlsym@@GLIBC_2.4'
external/__bazelrio_roborio_toolchain_linux/frc2022/roborio/arm-frc2022-linux-gnueabi/lib/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
As far as we can tell, this is an issue with dlsym not being present in the Linux simulation. Is this correct? How can we fix it?
BUILD file:
```
load("@bazelrio//:defs.bzl", "robot_cc_binary")

halsim_configs = {}

halsim_configs["gui"] = ["@bazelrio//libraries/cpp/wpilib/simulation/halsim_gui"]

halsim_configs["ds_socket"] = ["@bazelrio//libraries/cpp/wpilib/simulation/halsim_ds_socket"]

halsim_configs["ws_client"] = ["@bazelrio//libraries/cpp/wpilib/simulation/halsim_ws_client"]

halsim_configs["ws_server"] = ["@bazelrio//libraries/cpp/wpilib/simulation/halsim_ws_server"]

halsim_configs["combo"] = [
    "@bazelrio//libraries/cpp/wpilib/simulation/halsim_gui",
    "@bazelrio//libraries/cpp/wpilib/simulation/halsim_ws_server",
]

robot_cc_binary(
    name = "robot",
    srcs = [
        "src/robot.cpp",
        "src/robot.hpp",
    ],
    halsim_configs = halsim_configs,
    team_number = 6341,
    deps = [
        "@bazelrio//libraries/cpp/wpilib/cameraserver",
        "@bazelrio//libraries/cpp/wpilib/wpilibc",
        "@bazelrio//libraries/cpp/ctre/phoenix",
        "@bazelrio//libraries/cpp/rev/revlib",
        "@firelib",
        "//src/subsystems",
    ],
    copts = [
        "-ldl",
        "-lz"
    ]
)
```
WORKSPACE file:
```
local_repository(
    name = "bazelrio",
    path = "../../bazelRio/bazelrio",
)

local_repository(
    name="firelib",
    path="../fireLib"
)

load("@bazelrio//:deps.bzl", "setup_bazelrio_dependencies")

setup_bazelrio_dependencies()

load("@bazelrio//:defs.bzl", "setup_bazelrio")

setup_bazelrio()
```
Thanks in advance!

**Comment by Andrew-Cua** [comment]
adding `linkopts = ['-ldl']` to [https://github.com/bazelRio/bazelRio/blob/main/bazelrio/libraries/cpp/rev/revlib/BUILD#L1](https://github.com/bazelRio/bazelRio/blob/main/bazelrio/libraries/cpp/rev/revlib/BUILD#L1) fixed the issue for us.

**Comment by connorworley** [comment]
Please make a PR for the fix!

**Comment by Andrew-Cua** [comment]
Done!

**Comment by noamzaks** [comment]
Closed by #111
